### PR TITLE
8307865: Invalid is_in_any_VTMS_transition() check in post_dynamic_code_generated_while_holding_locks

### PR DESCRIPTION
**Pull request by lmesnik**
Method post_dynamic_code_generated_while_holding_locks()
register stubs and might be called during VTMT transitions.
At least it is called in tmp VTMT transition, and stubs might be generated during standard VTMT transition.

The method doesn't post event but just register stub for later posting so it might be called during transition.

Also, the test has been updated to test virtual threads. It crashed before fix and start passing after fix.
Additionally, checked this test with Xcomp, run tier1/tier5 and some stress testing

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307865](https://bugs.openjdk.org/browse/JDK-8307865): Invalid is_in_any_VTMS_transition() check in post_dynamic_code_generated_while_holding_locks

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [7f272622](https://git.openjdk.org/jdk/pull/13921/files/7f2726229e0d073c74f1ece7a1de67423ac633d1)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13921/head:pull/13921` \
`$ git checkout pull/13921`

Update a local copy of the PR: \
`$ git checkout pull/13921` \
`$ git pull https://git.openjdk.org/jdk.git pull/13921/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13921`

View PR using the GUI difftool: \
`$ git pr show -t 13921`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13921.diff">https://git.openjdk.org/jdk/pull/13921.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13921#issuecomment-1543029858)

### CODE REVIEW

**Review by sspitsyn — APPROVED** [review]
Looks good.
Thanks,
Serguei

File: test/hotspot/jtreg/serviceability/jvmti/DynamicCodeGenerated/DynamicCodeGeneratedTest.java [hunk 1]
```diff
@@ -49,14 +50,20 @@ public static void main(String[] args) throws Exception {
         threadChangeENM.setDaemon(true);
         threadChangeENM.start();
 
+        Runnable task = () -> {
+            String result = "string" + System.currentTimeMillis();
+            LockSupport.parkNanos(1);
```
[the remark this answers is not in the document]
Comment (lmesnik) [reply test/hotspot/jtreg/serviceability/jvmti/DynamicCodeGenerated/DynamicCodeGeneratedTest.java:55]: It is needed to provoke thread re-mounting to trigger VTMT transitions.
The test wouldn't fail without it.

[the remark this answers is not in the document]
Comment (plummercj) [reply test/hotspot/jtreg/serviceability/jvmti/DynamicCodeGenerated/DynamicCodeGeneratedTest.java:55]: I think you should add a comment for that.

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**